docker_cli/cli/setup_commands.py

```diff
@@ -0,0 +1,237 @@
+    -------
+    parser : argparse.ArgumentParser
+        The parser to add setup commands to.
+    prefix : str
+        The image tag prefix.
+    """
+
+    # Additional parsers for shared commands
+    setup_parse = argparse.ArgumentParser(add_help=False)
+    setup_parse.add_argument(
+        "--base",
+        "-b",
+        type=str,
+        required=True,
+        help="The name of the base Docker image.",
+    )
+
+    no_cache_parse = argparse.ArgumentParser(add_help=False)
+    no_cache_parse.add_argument(
+        "--no-cache",
+        action="store_true",
+        help="Run Docker build with no cache if used.",
+    )
+
+    cuda_run_parse = argparse.ArgumentParser(add_help=False)
+    cuda_run_parse.add_argument(
+        "--cuda-version",
+        "-c",
+        default="11.4",
+        type=str,
+        help="The CUDA version.",
+        metavar="VERSION",
+    )
+    cuda_run_parse.add_argument(
+        "--cuda-repo",
+        default="rhel8",
+        type=str,
+        help="The name of the CUDA repository for this distro "
+        '(e.g. "rhel8", "ubuntu2004".)',
+        metavar="REPO_NAME",
+    )
+
+    setup_parser = subparsers.add_parser(
+        "setup", help="Docker image setup commands.", formatter_class=help_formatter
+    )
+
+    setup_subparsers = setup_parser.add_subparsers(
+        dest="setup_subcommand", required=True
+    )
+
+    setup_all_parser = setup_subparsers.add_parser(
+        "all",
+        parents=[cuda_run_parse, no_cache_parse],
+        help="Set up the full Docker image stack.",
+        formatter_class=help_formatter,
+    )
+    setup_all_parser.add_argument(
+        "--tag",
+        "-t",
+        default="setup",
+        type=str,
+        help="The sub-prefix of the Docker images to be created. Generated images will "
+        f'have tags: "{prefix}-[TAG]-*".',
+    )
+    setup_all_parser.add_argument(
+        "--base",
+        "-b",
+        default="oraclelinux:8.4",
+        type=str,
+        help="The name of the parent Docker image.",
+    )
+    setup_all_parser.add_argument(
+        "--runtime-env-file",
+        default=run_reqs_path_str,
+        type=str,
+        help="The location of the runtime requirements file. Can be a pip-style "
+        "requirements.txt file, a conda-style environment.yml file, or a lockfile.",
+    )
+    setup_all_parser.add_argument(
+        "--dev-env-file",
+        default=dev_reqs_path_str,
+        type=str,
+        help="The location of the dev requirements file. Can be a pip-style "
+        "requirements.txt file, a conda-style environment.yml file, or a lockfile.",
+    )
+
+    setup_init_parser = setup_subparsers.add_parser(
+        "init",
+        parents=[no_cache_parse],
+        help="Set up the configuration image.",
+        formatter_class=help_formatter,
+    )
+    setup_init_parser.add_argument(
+        "--base",
+        "-b",
+        default="oraclelinux:8.4",
+        type=str,
+        required=True,
```

Comment:
Don't think it makes sense to mark this argument as required if it has a default value

```suggestion
```